src/azul/service/repository_controller.py

```diff
@@ -51,13 +55,18 @@ def download_file(self,
         plugin = self.repository_plugin(catalog)
         download_cls = plugin.file_download_class()
 
+        is_completed_range_request = False
+
         if request_index == 0:
             file = self.service.get_data_file(catalog=catalog,
                                               file_uuid=file_uuid,
                                               file_version=file_version)
             if file is None:
                 raise NotFoundError(f'Unable to find file {file_uuid!r}, '
                                     f'version {file_version!r} in catalog {catalog!r}')
+            match = self.ranged_request_re.fullmatch(headers.get('range', ''))
```

Comment:
You should parse this manually, in order to be stricter. If the header value starts with `bytes=` it would be an error for the remainder to not follow the syntax defined in the RFC: https://tools.ietf.org/html/rfc7233#section-2.1

For why, read https://en.wikipedia.org/wiki/Robustness_principle with special focus on the criticisms.

So partition on '=', check if LHS is `bytes`, then split on `,` if less than one entry, error, if more then one entry proceed normally, otherwise split that entry on `-`. If more or less than one entry, error, otherwise check if either part is an integer or empty string. If not, error, otherwise if first part is equal to the end of file and second part is empty, perform optimization, otherwise proceed normally. This is a pretty deep decision tree but it is an accurate implementation of the standard.